pkg/datapath/linux/node.go

```diff
@@ -903,7 +905,12 @@ func (n *linuxNodeHandler) nodeUpdate(oldNode, newNode *nodeTypes.Node, firstAdd
 	if newNode.IsLocal() {
 		isLocalNode = true
 		if n.nodeConfig.EnableLocalNodeRoute {
-			n.updateOrRemoveNodeRoutes([]*cidr.CIDR{oldIP4Cidr}, []*cidr.CIDR{newNode.IPv4AllocCIDR}, isLocalNode)
+			// Build a full list of CIDRs that are natively routed. We do not
+			// worry about duplicates as these are handled by the underlying
+			// call to cidr.DiffCIDRLists in updateOrRemoveNodeRoutes.
+			oldIPv4CIDRs := append([]*cidr.CIDR{oldIP4Cidr}, oldIPv4NativeCIDRs...)
+			newIPv4CIDRs := append([]*cidr.CIDR{newNode.IPv4AllocCIDR}, newNode.IPv4NativeRoutingCIDRs...)
+			n.updateOrRemoveNodeRoutes(oldIPv4CIDRs, newIPv4CIDRs, isLocalNode)
```

Comment:
This doesn't make sense, `updateOrRemoveNodeRoutes()` inserts routes pointing to the `cilium_host` interface. We'll need to update `ip rules` as done by:
```
func (info *RoutingInfo) Configure(ip net.IP, mtu int, compat bool) error {
```